In [1]:
from __future__ import absolute_import
import aiida
from aiida import load_profile
load_profile()

# Import commonly used functionality
import numpy as np
from aiida import orm, engine, common
from aiida.plugins import WorkflowFactory
from aiida.orm import Code
import sisl

In [2]:
!reentry scan
#!verdi daemon start
!verdi daemon restart

Restarting the daemon... OK


In [7]:
GaAs_222_dir = '/home/aakhtar/Calculations/2020/siesta/GaAs/LDA-standard/conventional-222-DONE/GaAs-222-5276028.frontal2/'
GaAs_222_GaV_0e_dir = '/home/aakhtar/Calculations/2020/siesta/GaAs/LDA-standard/conventional-222-GaV-0e-DONE/GaAs-222-GaV-0e-5276023.frontal2/'
GaAs_222_GaV_1e_dir = '/home/aakhtar/Calculations/2020/siesta/GaAs/LDA-standard/conventional-222-GaV-1e-DONE/GaAs-222-GaV-1e-5276034.frontal2/'
GaAs_222_GaV_2e_dir = '/home/aakhtar/Calculations/2020/siesta/GaAs/LDA-standard/conventional-222-GaV-2e-DONE/GaAs-222-GaV-2e-5276040.frontal2/'
GaAs_222_GaV_3e_dir ='/home/aakhtar/Calculations/2020/siesta/GaAs/LDA-standard/conventional-222-GaV-3e-DONE/GaAs-222-GaV-3e-5276041.frontal2/'
GaAs_222_GaV_1p_dir ='/home/aakhtar/Calculations/2020/siesta/GaAs/LDA-standard/conventional-222-GaV-1p-DONE/GaAs-222-GaV-1p-5276823.frontal2/'
#GaAs_222_GaV_2p_dir
#-------------
GaAs_333_dir = '/home/aakhtar/Calculations/2020/siesta/GaAs/LDA-standard/conventional-333-DONE/6213-farm2-node[49-52].out/'
GaAs_333_GaV_0e_dir = '/home/aakhtar/Calculations/2020/siesta/GaAs/LDA-standard/conventional-333-GaV-0e-DONE/6232-farm2-node[49-52].out/'
GaAs_333_GaV_1e_dir ='/home/aakhtar/Calculations/2020/siesta/GaAs/LDA-standard/conventional-333-GaV-1e-DONE/6235-farm2-node[35-38].out/'
GaAs_333_GaV_2e_dir ='/home/aakhtar/Calculations/2020/siesta/GaAs/LDA-standard/conventional-333-GaV-2e-DONE/6234-farm2-node[39-42].out/'
GaAs_333_GaV_3e_dir ='/home/aakhtar/Calculations/2020/siesta/GaAs/LDA-standard/conventional-333-GaV-3e-DONE/6233-farm2-node[44-47].out/'

In [8]:
def Read_FDF_File(path_dir,fdf_name):
    """
    Read The FDF File
    """
    fdf = sisl.get_sile(path_dir+fdf_name+'.fdf')

    return

def ASE2Siesta(A):
    """
    Getting ASE Atom Object and Converted to sisl
    """
    import sisl
    geo=sisl.Geometry.fromASE(A)
    return geo

In [9]:
def Siesta2ASE(spath,fdf_name):
    """
    Getting XV Path Object and Converted ASE Atom 
    """
    import sisl
    
    
    fdffile = sisl.get_sile(spath+"/"+fdf_name+".fdf")
    geometry = fdffile.read_geometry()
    
    ase_geom = sisl.Geometry.toASE(geometry)
    
    return ase_geom
 

In [10]:
GaAs_222 = Siesta2ASE(GaAs_222_dir,'input')
GaAs_222_GaV_0e = Siesta2ASE(GaAs_222_GaV_0e_dir,'input')
GaAs_222_GaV_1e = Siesta2ASE(GaAs_222_GaV_1e_dir,'input')
GaAs_222_GaV_2e = Siesta2ASE(GaAs_222_GaV_2e_dir,'input')
GaAs_222_GaV_3e = Siesta2ASE(GaAs_222_GaV_3e_dir,'input')
GaAs_333 = Siesta2ASE(GaAs_333_dir,'input')
GaAs_333_GaV_0e = Siesta2ASE(GaAs_333_GaV_0e_dir,'input')
GaAs_333_GaV_1e = Siesta2ASE(GaAs_333_GaV_1e_dir,'input')
GaAs_333_GaV_2e = Siesta2ASE(GaAs_333_GaV_2e_dir,'input')
GaAs_333_GaV_3e = Siesta2ASE(GaAs_333_GaV_3e_dir,'input')

In [11]:
print ('Pure Cell :'+str(GaAs_222.cell))
print ('Defect q=0 Cell :'+str(GaAs_222_GaV_0e.cell))
print ('Defect q=1 Cell :'+str(GaAs_222_GaV_1e.cell))
print ('Defect q=2 Cell :'+str(GaAs_222_GaV_2e.cell))
print ('Defect q=3 Cell :'+str(GaAs_222_GaV_3e.cell))
print ('Pure Cell :'+str(GaAs_333.cell))
print ('Defect q=0 Cell :'+str(GaAs_333_GaV_0e.cell))
print ('Defect q=1 Cell :'+str(GaAs_333_GaV_1e.cell))
print ('Defect q=2 Cell :'+str(GaAs_333_GaV_2e.cell))
print ('Defect q=3 Cell :'+str(GaAs_333_GaV_3e.cell))

Pure Cell :Cell([11.239312, 11.239314, 11.239314])
Defect q=0 Cell :Cell([11.239312, 11.239314, 11.239314])
Defect q=1 Cell :Cell([11.239312, 11.239314, 11.239314])
Defect q=2 Cell :Cell([11.239312, 11.239314, 11.239314])
Defect q=3 Cell :Cell([11.239312, 11.239314, 11.239314])
Pure Cell :Cell([16.858968, 16.858971, 16.858971])
Defect q=0 Cell :Cell([16.858968, 16.858968, 16.858968])
Defect q=1 Cell :Cell([16.858968, 16.858968, 16.858968])
Defect q=2 Cell :Cell([16.858968, 16.858968, 16.858968])
Defect q=3 Cell :Cell([16.858968, 16.858968, 16.858968])


In [12]:
from aiida_defects.formation_energy_siesta.formation_energy_siesta_manual import FormationEnergyWorkchainSIESTAManual
#import .formation_energy_siesta
#!pwd

In [13]:
from aiida.orm import StructureData
pure = StructureData(ase=GaAs_222)
defect_neutral = StructureData(ase=GaAs_222_GaV_0e)
defect_charge = StructureData(ase=GaAs_222_GaV_1e)
charge=-1
#defect_charge = StructureData(ase=GaAs_222_GaV_2e)
#defect_charge = StructureData(ase=GaAs_222_GaV_3e)
#-------------------------------------------------
#
#pure = StructureData(ase=GaAs_333)
#defect_neutral = StructureData(ase=GaAs_333_GaV_0e)
#defect_charge = StructureData(ase=GaAs_333_GaV_1e)
#defect_charge = StructureData(ase=GaAs_333_GaV_2e)
#defect_charge = StructureData(ase=GaAs_333_GaV_3e)



In [14]:
from aiida.orm import RemoteData
code = Code.get_from_string('siesta-psml-lua@pwk')
#--------------------------
remote_host = RemoteData(remote_path=GaAs_222_dir, computer=code.computer)
defect_neutral_remote = RemoteData(remote_path=GaAs_222_GaV_0e_dir, computer=code.computer)
defect_charged_remote = RemoteData(remote_path=GaAs_222_GaV_1e_dir, computer=code.computer)
#defect_charged_remote = RemoteData(remote_path=GaAs_222_GaV_2e_dir, computer=code.computer)
#defect_charged_remote = RemoteData(remote_path=GaAs_222_GaV_3e_dir, computer=code.computer)
#--------------------------
#remote_host = RemoteData(remote_path=GaAs_333_dir, computer=code.computer)
#defect_neutral_remote = RemoteData(remote_path=GaAs_333_GaV_0e_dir, computer=code.computer)
#defect_charged_remote = RemoteData(remote_path=GaAs_333_GaV_1e_dir, computer=code.computer)
#defect_charged_remote = RemoteData(remote_path=GaAs_333_GaV_2e_dir, computer=code.computer)
#defect_charged_remote = RemoteData(remote_path=GaAs_333_GaV_3e_dir, computer=code.computer)

In [17]:
inputs = {
    
    # Structures
    'host_structure': pure,
    'defect_structure': defect_neutral,
    'defect_charge' : defect_charge,
    'add_or_remove':orm.Str('remove'),
    'host_remote' : remote_host,
    'defect_neutral_remote': defect_neutral_remote,
    'defect_charged_remote': defect_charged_remote,
    #'host_node':orm.Int(19201),    # LAPTOP 20618
    #'defect_q0_node':orm.Int(19202),  # LAPTOP 20619
    #'defect_q_node':orm.Int(19203),  # LAPTOP 20620
    # Defect information 
    'defect_charge' : orm.Float(charge),
    'defect_site' : orm.List(list=[0.0,0.0,0.0]), #orm.List(list=[3.3226073300107264    , 1.918307998081692 ,17.5]),    # Position of the defect in crystal coordinates
    'fermi_level' : orm.Float(0.0),               # Position of the Fermi level, with respect to the valence band maximum      
    'chemical_potential' : orm.Float(-2077.548690 ), # eV, the chemical potentical of a O atom
    #'sigma':orm.Float(2.614),
    'use_siesta_mesh_cutoff':orm.Bool(True),
    'correction_scheme' : orm.Str('gaussian-model'), # gaussian-model ,gaussian , point, none
    "epsilon":orm.Float(12.4),
    #"pseudos_host":pseudos_dict_host,
    #"pseudos_defect":pseudos_dict_defect,
    #"pseudos_q0":pseudos_dict,
    #"pseudos_q":pseudos_dict,
    # Computational (chosen code is QE)
    #'siesta' : { 'dft': {'supercell_host':{'code': code, 'kpoints': kpoints_host, 'parameters' : parameters_host,
    #                                       'options':options_host,"basis": basis_dict_host},
    #                     'supercell_defect_q0':{'code': code, 'kpoints': kpoints_defect_q,'parameters' : parameters_defect_q0,
    #                                            'options':options_defect_q0,"basis": basis_dict_defect_q0},
    #                     'supercell_defect_q':{'code': code, 'kpoints': kpoints_defect_q,'parameters' : parameters_defect_q,
    #                                           'options':options_defect_q,"basis": basis_dict_defect_q}
}#}}

In [18]:
inputs

{'host_structure': <StructureData: uuid: 51ebebee-7b96-4d0c-97bc-239abfccc441 (unstored)>,
 'defect_structure': <StructureData: uuid: f662b8f9-8dc7-443d-b35f-75ca3e62bb9b (unstored)>,
 'defect_charge': <Float: uuid: e311460f-36b9-483a-af86-ac825c89fd64 (unstored) value: -1.0>,
 'add_or_remove': <Str: uuid: f2349737-ac32-44ae-b413-2a0054d3faf2 (unstored) value: remove>,
 'host_remote': <RemoteData: uuid: aeaa0461-8ed1-41e5-85e9-3e4554914493 (unstored)>,
 'defect_neutral_remote': <RemoteData: uuid: 1a2b71d6-6bb2-4239-b429-8dd6a4c81acd (unstored)>,
 'defect_charged_remote': <RemoteData: uuid: a09190c2-d3b5-4c66-942b-31215f67e5b9 (unstored)>,
 'defect_site': <List: uuid: 8e5d32d1-6669-4354-a9a6-93f993b1c6d6 (unstored) value: [0.0, 0.0, 0.0]>,
 'fermi_level': <Float: uuid: dcc0fc5d-a80c-4366-805d-7b21d64d2c0c (unstored) value: 0.0>,
 'chemical_potential': <Float: uuid: 4c174bd5-40f8-4675-a68e-d5c8349481c6 (unstored) value: -2077.54869>,
 'use_siesta_mesh_cutoff': <Bool: uuid: c24f6527-fae5-

In [19]:
workchain_future = engine.submit(FormationEnergyWorkchainSIESTAManual, **inputs)

In [20]:
workchain_future.pk

22378

pk=22378   GaAs 222 GaV q=-1

pk=   GaAs 222 GaV q=-2

pk=   GaAs 222 GaV q=-3

pk=   GaAs 333 GaV q=-1

pk=   GaAs 333 GaV q=-2

pk=   GaAs 333 GaV q=-3

pk=21662   GaAs 222 GaV q=-1

pk=21782   GaAs 222 GaV q=-2

pk=21898   GaAs 222 GaV q=-3

pk=22018   GaAs 333 GaV q=-1

pk=22138   GaAs 333 GaV q=-2

pk=22258   GaAs 333 GaV q=-3

In [22]:
%aiida

UsageError: Line magic function `%aiida` not found.


In [23]:
from aiida.manage.database.delete.nodes import delete_nodes


In [ ]:
delete_nodes()

In [27]:
from aiida.orm import QueryBuilder
from aiida.orm.nodes import ArrayData

In [28]:
qb = QueryBuilder()

qb.append(ArrayData)

qb.all()

[[<KpointsData: uuid: 69b2e40b-a138-4597-a9b6-4690f581d958 (pk: 16)>],
 [<KpointsData: uuid: 172bf1f6-080b-4bc4-91f4-94affc749b08 (pk: 31)>],
 [<KpointsData: uuid: a8b365cc-7853-4fb9-b925-301313747d4f (pk: 49)>],
 [<KpointsData: uuid: a720b2ad-d999-4dc3-8c11-01c5adc69785 (pk: 59)>],
 [<ArrayData: uuid: f8a93c2d-75cb-49fa-a743-27de4a6a6486 (pk: 65)>],
 [<ArrayData: uuid: c99795a7-fdf7-45b4-82ae-f6d57c06887c (pk: 86)>],
 [<KpointsData: uuid: 7660fb4e-972c-42d0-ae89-6edeb0386ba8 (pk: 75)>],
 [<ArrayData: uuid: 8ff2e4c7-b725-4f42-9e56-3c42c7bc7254 (pk: 81)>],
 [<KpointsData: uuid: 5adc2234-3f3a-4cdb-8e42-db0eca0e75be (pk: 99)>],
 [<KpointsData: uuid: 78306de8-d5c3-40d8-a312-d41fe2fbfd8f (pk: 116)>],
 [<KpointsData: uuid: 535cd06d-1919-42f4-bd3e-900d7ab47749 (pk: 106)>],
 [<KpointsData: uuid: 38f94845-eacc-4f3c-9f2f-c816f6968412 (pk: 132)>],
 [<KpointsData: uuid: f0bd4a8c-70ff-4402-b10d-65df4d5e0594 (pk: 124)>],
 [<KpointsData: uuid: 63def49f-3509-41f1-99de-2b0a5213c5db (pk: 140)>],
 [<Arra

In [36]:
qb.queryhelp

{'path': [{'entity_type': 'data.array.ArrayData.',
   'tag': 'ArrayData_1',
   'joining_keyword': None,
   'joining_value': None,
   'outerjoin': False,
   'edge_tag': None}],
 'filters': {'ArrayData_1': {'node_type': {'like': 'data.array.%'}}},
 'project': {'ArrayData_1': []},
 'order_by': {},
 'limit': None,
 'offset': None}

In [46]:
#qb.add_filter('ArrayData_1',{'pk':{'>':13000}})
qb.add_filter('filter':{'ArrayData_1': {'node_type': {'like': 'data.array.%'}}})

SyntaxError: invalid syntax (<ipython-input-46-e0fff9295fb2>, line 2)

In [43]:
qb.count

<bound method QueryBuilder.count of <aiida.orm.querybuilder.QueryBuilder object at 0x7f8038bb51c0>>

In [ ]:
def get_vbm_siesta_manual(remote_node,host_label):
    """
    Calculating valence band maximum from siesta calculations"
    """
    import sisl

    EIG = sisl.get_sile(remote_node.get_remote_path()+"/"+host_label+".EIG")

    N_electron = int(EIG.file.read_text().split()[3])
    vb_index = int(N_electron/2)-1

    eig_gamma=EIG.read_data()[0][0]

    vbm = np.amax(eig_gamma[vb_index])

    return vbm


In [ ]:
get_vbm_siesta_manual(remote_host,host_label='GaAs')

In [ ]:
import sisl
host_label='GaAs'
#EIG = sisl.get_sile(remote_host.get_remote_path()+"/"+host_label+".EIG")
BANDS = sisl.get_sile(remote_host.get_remote_path()+"/"+host_label+".bands")

In [ ]:
def get_vbm_siesta_manual_bands(remote_node,host_label,NE):
    """
    Calculating valence band maximum from siesta calculations"
    """
    import sisl

    BANDS = sisl.get_sile(remote_node.get_remote_path()+"/"+host_label+".bands")

    #N_electron = int(BANDS.file.read_text().split()[3])
    N_electron = NE
    vb_index = int(N_electron/2)-1

    eig_gamma=BANDS.read_data()

    vbm = np.amax(eig_gamma[2][0][0][vb_index])
    #print(vbm)
    return vbm


In [ ]:
def get_vbm_siesta_manual_bands_normal(dir_path,host_label,NE):
    """
    Calculating valence band maximum from siesta calculations"
    """
    import sisl

    BANDS = sisl.get_sile(dir_path+"/"+host_label+".bands")

    #N_electron = int(BANDS.file.read_text().split()[3])
    N_electron = NE
    vb_index = int(N_electron/2)-1

    eig_gamma=BANDS.read_data()

    vbm = np.amax(eig_gamma[2][0][0][vb_index])
    #print(vbm)
    return vbm


In [ ]:
GaAS_total_electrons_222=896
GaAs_GaV_0e_total_electrons_222 =883.000000
GaAs_GaV_1e_total_electrons_222=884.000000
GaAs_GaV_2e_total_electrons_222=885.000000
GaAs_GaV_3e_total_electrons_222=886.000000
GaAs_total_electrons_333=3024
GaAs_GaV_0e_total_electrons_333 =3011.000000
GaAs_GaV_1e_total_electrons_333=3012.000000
GaAs_GaV_2e_total_electrons_333=3013.000000
GaAs_GaV_3e_total_electrons_333=3014.000000
GaAS_AllE_444=7156
get_vbm_siesta_manual_bands_normal('/home/aakhtar/Calculations/2020/siesta/GaAs/LDA-standard/conventional-222-GaV-0e-DONE/GaAs-222-GaV-0e-5276023.frontal2',
                                   'GaAs',
                                   GaAs_GaV_0e_total_electrons_222)

In [ ]:
remote_host.get_remote_path()

In [ ]:
N_electron = 896

In [ ]:
get_vbm_siesta_manual_bands(remote_host,host_label,N_electron)

In [ ]:
vb_index = int(N_electron/2)-1
vb_index

In [ ]:
#eig_gamma=EIG.read_data()[0][0]
eig_gamma=BANDS.read_data()
eig_gamma

In [ ]:
eig_gamma[0][1]

In [ ]:
eig_gamma[2].shape

In [ ]:
for i in range(551):
    print (eig_gamma[2][i][0][-1])

In [ ]:
#vbm = np.amax(eig_gamma[vb_index])
vbm = np.amax(eig_gamma[0][0][0][vb_index])
vbm

In [ ]:
EIG.read_data()

In [ ]:
EIG.read_fermi_level()

In [ ]:
def output_total_electrons_manual(Remote_node):
    """
    Return Number of Electrons in System from output.out file 
    """
    import sisl
    out = sisl.io.outSileSiesta(Remote_node.get_remote_path()+"/output.out")
    f=open(out.file)
    a=f.readlines()
    for line in range(len(a)):
        if 'Total number of electrons:' in a[line]:
            number_of_electrons = int(float(a[line].split()[4]))
            #print  (a[line].split())
    return number_of_electrons


In [ ]:
output_total_electrons_manual(remote_host)

In [ ]:
int(float(output_total_electrons_manual(remote_host)))

In [ ]:
from aiida.orm import load_node
results =load_node('20241') #20043

In [ ]:
results.get_remote_path

In [ ]:
fdf_file = sisl.get_sile(results.get_remote_path()+"/input.fdf")

In [ ]:
out = sisl.io.outSileSiesta(results.get_remote_path()+"/output.out")

In [ ]:
f=open(out.file)

In [ ]:
a=f.readlines()

In [ ]:
len(a)

In [ ]:
a[1207]

In [ ]:
def OutputEnergyManual(Remote_node):
    """
    
    """
    from aiida.orm import Float
    out = sisl.io.outSileSiesta(Remote_node.get_remote_path()+"/output.out")
    f=open(out.file)
    a=f.readlines()
    for line in range(len(a)):
        if 'siesta:         Total =' in a[line]:
            energy = orm.Float(a[line].split()[3])
            #print 
    return energy

In [ ]:
OutputEnergyManual(results)

In [ ]:
f.close()

In [ ]:
for line in a:
    print (line)

In [ ]:
f= results.get_remote_path+"/output.out"
open f as

In [ ]:
a=out.read()

In [ ]:
a

In [ ]:
fdf_file.get("SystemLAbel")

In [ ]:
fdf_file.read_fermi_level()

In [ ]:
data_array = sisl.get_sile(remote_pure_results.get_remote_path()+"/GaAs.VT")


In [ ]:
vt = data_array.read_grid()
vt_data = orm.ArrayData()
vt_data_grid = orm.ArrayData()
vt_data.set_array('data', vt.grid) # The array
vt_data_grid.set_array('grid',np.array(vt.shape)) # The Grid Shape

In [ ]:
from aiida.orm import SinglefileData

In [ ]:
VT = SinglefileData(GaAs_222_dir +'/GaAs.VT',computer=code.computer)

In [ ]:
VT.store()

In [ ]:
from aiida.orm import load_node
remote_pure_results =load_node('20049') #

In [ ]:
remote_pure_results.

In [ ]:
data_array = sisl.get_sile(remote_pure_results.filename)

In [ ]:
data_array.read_grid()

In [ ]:
VT.filename

In [ ]:
Read_FDF_File(GaAs_222_dir,'input')

In [ ]:
VT= sisl.get_sile()

In [ ]:
from aiida.orm import load_node
results_VT=load_node('20042')

In [ ]:
results_VT

In [ ]:
data_array = sisl.get_sile(results_VT.filename)


In [ ]:
data_array.read_grid()

In [ ]:
vt = data_array.read_grid


In [ ]:
vt.gr

In [ ]:
data_array = sisl.get_sile(results_VT.filename)
vt = data_array.read_grid()
vt_data = orm.ArrayData()
vt_data_grid = orm.ArrayData()
vt_data.set_array('data', vt.grid) # The array
vt_data_grid.set_array('grid',np.array(vt.shape)) # The Grid Shape

#self.ctx.v_host = vt_data
#self.ctx.v_host_grid = vt_data_grid
#self.report("DEBUG: The host VT file Read!")
#self.report("DEBUG: VT Array = "+str(self.ctx.v_host ))
#self.report("DEBUG: VT Array Shape = "+str(self.ctx.v_host_grid.get_array('grid')))
#self.ctx.host_vbm = orm.Float(get_vbm_siesta(host_vt))
#self.report("DEBUG: VBM = "+str(self.ctx.host_vbm.value))


In [ ]:
#results.output.output_parameters.get_dict()

In [ ]:
#results

In [ ]:
from qe_tools import CONSTANTS

In [ ]:
CONSTANTS.

In [ ]:
#!verdi process show 3160

In [ ]:
#!verdi process report 3160

In [ ]:
#workchain_future.outputs.formation_energy_uncorrected.value #.outputs.formation_energy_uncorrected.value
#from aiida.orm import load_node
#results=load_node('20618')
#import sisl
#data_array = sisl.get_sile(results.get_remote_workdir()+"/aiida.VT")
#vt = data_array.read_grid()

In [ ]:
#import numpy as np
#vt_test = np.array(vt.shape)

In [ ]:
#vt_test

In [ ]:
#vt_grid = vt.shape

In [ ]:
#vt_grid

In [ ]:
#results.get_remote_workdir()

In [ ]:
#results.outputs['output_parameters'].get_dict()['mesh']
#results['forces_and_stress']

In [ ]:
#results.outputs.output_parameters.get_dict()

In [ ]:
#code

In [ ]:
#from aiida_siesta.calculations.siesta import SiestaCalculation
#builder = SiestaCalculation.get_builder()

In [ ]:
#builder.metadata

In [ ]:
#import sisl
#fdf=sisl.get_sile("/home/aakhtar/aiida_scartch/45/17/087b-33b2-42a2-8f27-185b3bc39cad/aiida.fdf") 
#out=sisl.get_sile("/home/aakhtar/aiida_scartch/00/9b/e245-59f6-4638-8725-a4fced1dd07e/aiida.out") 
#a=out.readline()

In [ ]:
#test=fdf.read_geometry()
#data_array = sisl.get_sile("/home/aakhtar/aiida_scartch/00/9b/e245-59f6-4638-8725-a4fced1dd07e/aiida.EIG") 
#data_array = sisl.get_sile("/home/aakhtar/aiida_scartch/83/7d/dd23-b492-4eb2-a266-99260ad907ee/aiida.EIG") 


In [ ]:

#data_array.file.read_text()

In [ ]:
#data_array.read_data()[0][0]

In [ ]:
#N_electron=int(data_array.file.read_text().split()[3])
#vb_index = int(N_electron/2)-1
#vb_index

In [ ]:
#fermi=data_array.read_fermi_level()
#eig=data_array.read_data()[0][0]
#print (fermi)

In [ ]:
#import numpy as np

In [ ]:
#for i in eig:
#    print (i)

In [ ]:
#vbm = np.amax(eig[vb_index])

# Model Charge

In [ ]:
#data_array.read_data()

In [ ]:
#data_array = sisl.get_sile("/home/aakhtar/aiida_scartch/f1/42/f7ea-5d86-40a8-9963-2ec31bfb426a/aiida.VT") 

In [ ]:
#vt=data_array.read_grid()

In [ ]:
#vt.shape

In [ ]:
#get_reciprocal_grid(test.rcell,50.0)

In [ ]:
#get_model_potential_with_siesta_mesh()
#from aiida.orm import load_node
#results=load_node('9478')


In [ ]:
#a=results.inputs.structure

In [ ]:
#import sisl
#fdf=sisl.get_sile("/home/aakhtar/aiida_scartch/45/17/087b-33b2-42a2-8f27-185b3bc39cad/aiida.fdf") 
#test=fdf.read_geometry()
#test.rcell